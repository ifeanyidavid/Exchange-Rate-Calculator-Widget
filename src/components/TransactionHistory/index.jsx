import React from "react";
import PropTypes from "prop-types";
import DateComponent from "../DateComponent";
import TransactionHistoryItem from "../TransactionHistoryItem";

function TransactionHistory({ transactionHistory }) {
  const marginBottom = transactionHistory.length > 1 ? "mb-2" : "mb-0";
  return (
    <>
      <DateComponent />
      {transactionHistory.map(history => (
        <TransactionHistoryItem
          key={history.id}
          history={history}
          hasMarginBottom={marginBottom}
        />
      ))}
    </>
  );
}

export default TransactionHistory;

TransactionHistory.propTypes = {
  transactionHistory: PropTypes.array
};
