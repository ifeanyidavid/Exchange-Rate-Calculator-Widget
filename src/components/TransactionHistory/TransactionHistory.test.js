import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import TransactionHistory from ".";

configure({ adapter: new Adapter() });

describe("TransactionHistory component", () => {
  it("Renders correctly", () => {
    const transactionHistory = [
      {
        id: 0,
        description: `Exchanged to EUR`,
        time: new Date().toLocaleTimeString("en-GB"),
        sourceAmount: 23,
        destinationAmount: 42,
        sourceCurrency: "USD",
        destinationCurrency: "EUR"
      }
    ];
    const props = {
      transactionHistory
    };
    // wrapper instance around rendered output
    const wrapper = shallow(<TransactionHistory {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
