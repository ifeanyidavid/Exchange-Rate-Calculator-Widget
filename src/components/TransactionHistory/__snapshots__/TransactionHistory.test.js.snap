// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionHistory component Renders correctly 1`] = `
<Fragment>
  <DateComponent />
  <TransactionHistoryItem
    hasMarginBottom="mb-0"
    history={
      Object {
        "description": "Exchanged to EUR",
        "destinationAmount": 42,
        "destinationCurrency": "EUR",
        "id": 0,
        "sourceAmount": 23,
        "sourceCurrency": "USD",
        "time": "2:35:41 PM",
      }
    }
    key="0"
  />
</Fragment>
`;
